src/service/BaseService.ts

```diff
@@ -0,0 +1,54 @@
+import {apiUrl} from '../constants';
+import {Alert} from 'react-native';
+
+export default class BaseService {
+  request = async <T>({
+    endpoint,
+    token,
+    method = 'GET',
+    headers,
+    body,
+    errorText,
+  }: {
+    endpoint: string;
+    token?: string;
+    method?: Request['method'];
+    headers?: Request['headers'];
+    body?: Request['body'];
+    errorText?: string;
+  }): Promise<T | undefined> => {
+    try {
+      const authHeader = token && {
+        Authorization: `Bearer ${token}`,
+      };
+
+      const rawResponse = await fetch(`${apiUrl}${endpoint}`, {
+        method,
+        headers: {
+          ...headers,
+          ...authHeader,
+        },
+        body,
+      });
+
+      if (!rawResponse.ok) {
+        this.handleError(rawResponse); 
+      }
+
+      return await rawResponse.json();
+    } catch (err) {
+      Alert.alert('oh snap!', errorText || err);
```

Comment:
I know this must be just for demo purposes, but try to separate the service from the error handling/error showing

Reply:
Need clarification